text/0000-tracked-properties.md

```diff
@@ -227,34 +248,350 @@ memoization) if a property shows up as a bottleneck during profiling.
+just like our previous example:
+
+```js
+import { tracked } from '@ember/object';
+
+class Person {
+  @tracked firstName = 'Tom';
+  @tracked lastName = 'Dale';
+  middleName = 'Tomster';
+
+  @tracked
+  get fullName() {
+    return `${this.firstName} ${this.middleName} ${this.lastName}`;
+  }
+}
+```
+
+The last piece of the puzzle here will be a development time only assertion.
+When you access _untracked_ properties from an a tracked getter, we will install
+something similar to the mandatory setter function. This will throw an error if
+the user ever tries to set the property, effectively making it immutable.
+
+To accomplish this, we can wrap any object which has tracked properties with a
+native [`Proxy`][proxy], which will subsequently wrap any objects or array properties
+which are accessed with another proxy, and so on. These proxies will be able
+to check if any untracked properties are being accessed from an tracked context,
+and setup immutable assertions if so.
+
+### Tracked Interop with Computed Properties and `Ember.set`
+
+Tracked properties work well in isolation. In systems built from the ground up
+using `@tracked`, all mutable values will be marked as tracked, so autotracking
+will be able to "watch" them all as they are accessed and changed.
+
+However, tracked properties will have to interoperate with the current system
+for tracking changes and mutable state in Ember for some time as the community
+transitions forward. The origin of all mutations in the old system is through
+`Ember.set` (or more specifically, `notifyPropertyChange`), but there are many
+ways to listen to these changes. Tracked properties are concerned with two
+specific cases:
+
+1. **Computed Properties.** When `notifyPropertyChange` is called on one of a
+computed property's dependencies, it will invalidate, and invalidate any computed properties which
+depend on it. In this way, one call to `set` can invalidate many properties in
+the system. Tracked properties must also properly invalidate if they consume
+any of these properties.
+2. **Ember.set() directly**. If a tracked property accesses a simple, untracked
+property which is _later_ set with `Ember.set()`, it must have a way to know
+that this property has changed. This can occur in services, helper classes which
+have not yet converted to `@tracked`, and in many other edge cases.
+
+In the case of computed properties, we have all the information we need up front
+to allow tracked getters to watch for changes in the computed. They will notify
+whenever they are updated, so the same technique which is used for the autotrack
+stack can be used to watch them as well. Crucially, computed properties will
+always have a tag due to the fact that they are defined explicitly in class
+definitions, which means we have something to watch the very first time a
+computed property is accessed.
+
+In the edge case of non-computed, non-tracked property access, this is not the
```

Comment:
Two things here:

1. I don't think non-computed, non-tracked is really an edge case—it's quite common in most apps. Calling it an edge case here might raise some hackles, and it's not important to our argument how edge-y or not it is.
2. To me, the important aspect isn't "whether it has a tag," which is kind of orthogonal, but about whether the system _knows about/can intercept_ the property. Even if it didn't have a validator assigned, trapping unknown property access would be enough, if the language supported that capability.